include/seqan3/search/fm_index/bi_fm_index.hpp

```diff
@@ -55,15 +64,17 @@ namespace seqan3
  *
  * \attention When building an index for a **text collection** over any alphabet, the symbols with rank 254 and 255
  *            are reserved and may not be used in the text.
+ *
+ * \deprecated Use seqan3::text_layout to indicate single texts and text collections. The use of bool is deprecated.
  */
-template <bool is_collection = false, detail::SdslIndex sdsl_index_type_ = default_sdsl_index_type>
+template <auto is_collection = text_layout::single, detail::SdslIndex sdsl_index_type_ = default_sdsl_index_type>
 class bi_fm_index
 {
 protected:
     //!\brief The alphabet size of the text.
     size_t sigma{0};
     //!\brief Indicates whether index is built over a collection.
-    static constexpr bool is_collection_{is_collection};
+    static constexpr text_layout is_collection_{is_collection};
```

Comment:
This can stay the same so you don't need all the casts. The strong_type is just for protecting the interface...